addons/mail/static/tests/thread/unread_messages_banner.test.js

```diff
@@ -53,3 +65,97 @@ test("mark thread as read from unread messages banner", async () => {
     });
     await contains(".o-mail-Thread-jumpToUnread", { count: 0 });
 });
+
+test("scroll to the first unread message", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "general" });
+    const messageIds = [];
+    for (let i = 0; i < 201; i++) {
+        messageIds.push(
+            pyEnv["mail.message"].create({
+                body: `message ${i}`,
+                model: "discuss.channel",
+                res_id: channelId,
+            })
+        );
+    }
+    const [selfMember] = pyEnv["discuss.channel.member"].search([
+        ["partner_id", "=", serverState.partnerId],
+        ["channel_id", "=", channelId],
+    ]);
+    pyEnv["discuss.channel.member"].write([selfMember], { new_message_separator: messageIds[100] });
+    let slowRegisterMessageRef = false;
+    patchWithCleanup(Thread.prototype, {
+        async registerMessageRef() {
+            if (slowRegisterMessageRef) {
+                // Ensure scroll is made even when messages are mounted later.
+                await new Promise((res) => setTimeout(res, 500));
+            }
+            super.registerMessageRef(...arguments);
+        },
+    });
+    await start();
+    await openDiscuss(channelId);
+    await click(".o-mail-Thread-banner", {
+        text: "You're viewing older messagesJump to Present",
+    });
+    await scroll(".o-mail-Thread", "bottom");
+    await contains(".o-mail-Thread", { scroll: "bottom" });
+    slowRegisterMessageRef = true;
+    await click("span", {
+        text: "101 new messages",
```

Comment:
I would prefer another test for the `slow register` part, so that this is more advanced and it's clearly stated in the name of test